tket/src/Mapping/MappingFrontier.cpp

```diff
@@ -673,6 +679,16 @@ void MappingFrontier::add_bridge(
     central_in_it = this->linear_boundary->find(central);
   }
 
+  Node central_node = Node(central);
+  // In this case we can still use the node, but we need to make sure that
+  // the wire it's on won't be reassigned
+  // This could reduce performance, but the "reassignable_nodes_" case
+  // deals with Qubit wires that have no multi-qubit quantum operations,
+  // so in practice this won't change the results for a meaningful circuit
+  if (this->reassignable_nodes_.find(central_node) !=
+      this->reassignable_nodes_.end()) {
+    this->reassignable_nodes_.erase(central_node);
```

Comment:
Can we try to exercise this condition in a test?

Reply:
done